commands/operator-sdk/cmd/scorecard/scorecard.go

```diff
@@ -0,0 +1,195 @@
+	OlmTests       bool
+	NamespacedMan  string
+	GlobalMan      string
+	CrMan          string
+}
+
+var SCConf Config
+
+const (
+	basicOperator  = "Basic Operator"
+	olmIntegration = "OLM Integration"
+	goodTenant     = "Good Tentant"
+)
+
+type scorecardTest struct {
+	testType      string
+	name          string
+	description   string
+	earnedPoints  int
+	maximumPoints int
+}
+
+type cleanupFn func() error
+
+var (
+	apiversion     string
+	kind           string
+	name           string
+	kubeconfig     *rest.Config
+	scTests        []scorecardTest
+	dynamicDecoder runtime.Decoder
+	runtimeClient  client.Client
+	restMapper     *restmapper.DeferredDiscoveryRESTMapper
+	deploymentName string
+	proxyPod       *v1.Pod
+	cleanupFns     []cleanupFn
+)
+
+const scorecardPodName = "operator-scorecard-test"
+
+func ScorecardTests(cmd *cobra.Command, args []string) error {
+	// in main.go, we catch and print errors, so we don't want cobra to print the error itself
+	cmd.SilenceErrors = true
+	cmd.SilenceUsage = true
+	defer cleanupScorecard()
+	var err error
+	kubeconfig, SCConf.Namespace, err = k8sInternal.GetKubeconfigAndNamespace(SCConf.KubeconfigPath)
+	if err != nil {
+		return err
+	}
+	scheme := runtime.NewScheme()
+	// scheme for client go
+	cgoscheme.AddToScheme(scheme)
+	// api extensions scheme (CRDs)
+	extscheme.AddToScheme(scheme)
+	// olm api (CSVs)
+	olmApi.AddToScheme(scheme)
+	dynamicDecoder = serializer.NewCodecFactory(scheme).UniversalDeserializer()
+	// if a user creates a new CRD, we need to be able to reset the rest mapper
+	// temporary kubeclient to get a cached discovery
+	kubeclient, err := kubernetes.NewForConfig(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to get a kubeclient: %v", err)
+	}
+	cachedDiscoveryClient := cached.NewMemCacheClient(kubeclient.Discovery())
+	restMapper = restmapper.NewDeferredDiscoveryRESTMapper(cachedDiscoveryClient)
+	restMapper.Reset()
+	runtimeClient, _ = client.New(kubeconfig, client.Options{Scheme: scheme, Mapper: restMapper})
+	err = createFromYAMLFile(SCConf.GlobalMan, false)
+	if err != nil {
+		return fmt.Errorf("failed to create global resources: %v", err)
+	}
+	err = createFromYAMLFile(SCConf.NamespacedMan, false)
+	if err != nil {
+		return fmt.Errorf("failed to create namespaced resources: %v", err)
+	}
+	err = createFromYAMLFile(SCConf.CrMan, true)
+	if err != nil {
+		return fmt.Errorf("failed to create cr resource: %v", err)
+	}
+	obj := unstructured.Unstructured{}
+	obj.SetAPIVersion(apiversion)
+	obj.SetKind(kind)
+	if SCConf.BasicTests {
+		fmt.Printf("Checking spec and stat\n")
+		err = checkSpecAndStat(runtimeClient, obj)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Checking status update\n")
+		err = checkStatusUpdate(runtimeClient, obj)
+		if err != nil {
+			return err
+		}
+		logs, err := writingIntoCRsHasEffect(obj)
+		if err != nil {
+			return err
+		}
+
+		/* this print out a lot of ugly logs and should only be used for debugging
```

Comment:
Since we have different log levels now we can just use that? Otherwise we can remove it, wdyt?